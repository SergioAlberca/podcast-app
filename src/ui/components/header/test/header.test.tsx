import { render, screen } from "@/test/test_utils";
import Header from "../header_component";
import { BrowserRouter } from "react-router-dom";

describe("Header component tests", () => {
  it("render component", () => {
    const { container } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it("Podcaster test is in the document", () => {
    render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(screen.getByText("Podcaster")).toBeInTheDocument();
  });

  it("Loading is showing", () => {
    render(
      <BrowserRouter>
        <Header isLoadingRoute={true} />
      </BrowserRouter>
    );
    expect(screen.getByAltText("loading")).toBeInTheDocument();
  });
});
