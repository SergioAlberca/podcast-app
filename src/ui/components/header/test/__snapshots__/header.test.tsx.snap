// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component tests > render component 1`] = `
<div>
  <header
    class="_header_a4172c"
  >
    <a
      href="/"
    >
      <h2>
        Podcaster
      </h2>
    </a>
  </header>
</div>
`;
